src/actions/attack.cpp

```diff
@@ -1339,6 +1339,25 @@ void attack::perform()
 		return;
 	}
 
+	bc_.reset(new battle_context(units_, a_.loc_, d_.loc_, a_.weapon_, d_.weapon_));
```

Comment:
Shouldn't the `VALIDATE(a_.weapon_ < ...)` be moved up to be before this line?

Reply:
I guess no, because you can change the selected attack in the `pre_attack` event.